src/test/java/seedu/address/model/person/MatricNumberTest.java

```diff
@@ -0,0 +1,43 @@
+package seedu.address.model.person;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import seedu.address.testutil.Assert;
+
+public class MatricNumberTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        Assert.assertThrows(NullPointerException.class, () -> new MatricNumber(null));
+    }
+
+    @Test
+    public void constructor_invalidMatricNumber_throwsIllegalArgumentException() {
+        String invalidMatricNumber = "";
+        Assert.assertThrows(IllegalArgumentException.class, () -> new MatricNumber(invalidMatricNumber));
+    }
+
+    @Test
+    public void isValidMatricNumber() {
+        // null matric number
+        Assert.assertThrows(NullPointerException.class, () -> MatricNumber.isValidMatricNumber(null));
+
+        // invalid matric numbers
+        assertFalse(MatricNumber.isValidMatricNumber("")); // empty string
+        assertFalse(MatricNumber.isValidMatricNumber(" ")); // spaces only
+        assertFalse(MatricNumber.isValidMatricNumber("A1234567"));
+        assertFalse(MatricNumber.isValidMatricNumber("1234567A"));
+        assertFalse(MatricNumber.isValidMatricNumber("987645321"));
+
+
```

Comment:
if we're gonna stick with matric number starting with an 'A' and having a valid year, we'll have to change the test cases